src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -4412,11 +4412,10 @@ static void jumpSpace( const QString &txt, int &i )
 QString QgsPostgresProvider::getNextString( const QString &txt, int &i, const QString &sep )
 {
   jumpSpace( txt, i );
-  QString cur = txt.mid( i );
-  if ( cur.startsWith( '"' ) )
+  if ( i < txt.length() && txt.at( i ) == '"' )
   {
     QRegExp stringRe( "^\"((?:\\\\.|[^\"\\\\])*)\".*" );
```

Comment:
Would be interesting to see if upgrading this to QRegularExpression would also help

Reply:
Good point, I was not aware of this new class. I also wonder if storing the regexp statically would be better than constructing it each time ... but I am not sure.

Reply:
It gets a bit tricky there - neither of the qt regex class are thread safe, so I don't think a static one can be used here

Reply:
That's a pattern I've used (static regexp) until Nyall pointed out that `QgsRegularExpression` is not thread safe, so using a static may not be a good idea.

Reply:
Ah good point ! Thanks